airbyte-integrations/connectors/source-hubspot/source_hubspot/manifest.yaml

```diff
@@ -471,7 +585,290 @@ definitions:
+      paginator:
+        $ref: "#/definitions/cursor_paginator"
+      record_selector:
+        type: RecordSelector
+        extractor:
+          type: DpathExtractor
+          field_path:
+            - results
+    transformations:
+      - type: AddFields
+        fields:
+          - path: ["archivedAt"]
+            value: "{{ record.get('archivedAt') or record['createdAt'] }}"
+      - type: CustomTransformation
+        class_name: source_hubspot.components.NewtoLegacyFieldTransformation
+        field_mapping:
+          hs_date_entered_: "hs_v2_date_entered_"
+          hs_date_exited_: "hs_v2_date_exited_"
+          hs_time_in_: "hs_v2_latest_time_in_"
+      - type: DpathFlattenFields
+        field_path: ["properties"]
+        key_transformation:
+          type: KeyTransformation
+          prefix: properties_
+    incremental_sync:
+      $ref: "#/definitions/incremental_sync"
+      cursor_field: archivedAt
+      is_client_side_incremental: true
+    state_migrations:
+      - type: CustomStateMigration
+        class_name: source_hubspot.components.MigrateEmptyStringState
+        cursor_field: archivedAt
+    schema_loader:
+      - $ref: "#/definitions/deals_archived_schema_loader"
+      - type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/deals_archived"
+
+  forms_stream:
+    primary_key:
+      - id
+    $ref: "#/definitions/stream_base"
+    name: forms
+    retriever:
+      $ref: "#/definitions/base_retriever"
+      requester:
+        $ref: "#/definitions/base_requester"
+        path: /marketing/v3/forms
+        request_parameters:
+          formTypes:
+            "['hubspot', 'captured', 'flow']"
+            # Not supported by `v1` api version used by the FormSubmissions stream
+            # TODO: consider migrating FormSubmissions stream to use `/v3` api version
+            # then uncomment the last 2 form types.
+            # "blog_comment",
+            # "all",
+          properties:
+            type: QueryProperties
+            property_list:
+              type: PropertiesFromEndpoint
+              property_field_path:
+                - name
+              retriever:
+                type: SimpleRetriever
+                requester:
+                  $ref: "#/definitions/base_requester"
+                  path: /properties/v2/form/properties
+                decoder:
+                  type: JsonDecoder
+                record_selector:
+                  type: RecordSelector
+                  extractor:
+                    type: DpathExtractor
+                    field_path: []
+            property_chunking:
+              type: PropertyChunking
+              property_limit_type: characters
+              property_limit: 15000
+              record_merge_strategy:
+                type: GroupByKeyMergeStrategy
+                key: id
+      paginator:
+        $ref: "#/definitions/cursor_paginator"
+      record_selector:
+        type: RecordSelector
```

Comment:
wanted to note that when @darynaishchenko was implementing `goals`, she noticed that we aren't properly normalizing the properties values back to their expected types. She added a custom normalization strategy for this that I think we'll need to incorporate into some of these streams with dynamic properties. See here:

https://github.com/airbytehq/airbyte/pull/59727/files#diff-1b47d519451dbf69102c8e716c20ddbbc3049b92cde345bdca1a49757276583eR924-R928

Reply:
added custom schema normalization